api/src/glfs-fops.c

```diff
@@ -967,11 +967,14 @@ pub_glfs_fstat(struct glfs_fd *glfd, struct stat *stat)
         goto out;
     }
 
-    ret = syncop_fstat(subvol, fd, &iatt, NULL, NULL);
+    if (stat) {
+        ret = syncop_fstat(subvol, fd, &iatt, NULL, NULL);
+        if (ret == 0)
+            glfs_iatt_to_stat(glfd->fs, &iatt, stat);
+    } else
+        ret = syncop_fstat(subvol, fd, NULL, NULL, NULL);
```

Comment:
I don't think this serves any purpose if we are not getting the iatt data.

Reply:
I did not want to change too much existing functionality though.